sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/CommonUtil.kt

```diff
@@ -0,0 +1,246 @@
+    private val logger = Logger(CommonUtil::class.java.simpleName)
+
+    /**
+     * Get the encoded byte array of string from character set
+     *
+     * @param rawString the raw string.
+     * @param charsetName the character set, i.e. "UTF-8", "UTF-16"
+     * @return the ByteArray of the string
+     */
+    fun getEncodedByteArray(rawString: String, charsetName: String): ByteArray {
+        return rawString.toByteArray(Charset.forName(charsetName))
+    }
+
+    /**
+     * Get the encoded MD5 digest string
+     *
+     * @param rawString the raw string.
+     * @return the MD5 digest string
+     */
+    fun getMD5HashData(rawString: String?): String? {
+        return if (rawString == null) {
+            null
+        } else try {
+            String.format(
+                "%032x",
+                BigInteger(
+                    1,
+                    MessageDigest.getInstance("MD5")
+                        .digest(getEncodedByteArray(rawString, "UTF-8"))
+                )
+            )
+        } catch (swallowed: NoSuchAlgorithmException) {
+            null
+        }
+    }
+
+    /**
+     * Check whether the strings are same.
+     *
+     * @param first the first string
+     * @param second the second string
+     * @return true if same otherwise false
+     */
+    fun isEqual(first: String, second: String): Boolean {
+        return first == second
+    }
+
+    /**
+     * Check the string whether null or empty
+     *
+     * @param rawString the raw string
+     * @return true if null or empty otherwise false
+     */
+    fun isNullOrEmpty(rawString: String?): Boolean {
+        return rawString.isNullOrEmpty()
+    }
+
+    /**
+     * Get the encoded Sha256 digest string
+     *
+     * @param rawString the raw string.
+     * @return the encoded string
+     */
+    fun getSha256HashData(rawString: String?): String? {
+        return if (rawString == null) {
+            null
+        } else try {
+            val md = MessageDigest.getInstance("SHA-256")
+            val byteData = md.digest(getEncodedByteArray(rawString, "utf-8"))
+            val sb = StringBuilder()
+            for (b in byteData) {
+                val hex = Integer.toHexString(0xFF and b.toInt())
+                if (hex.length == 1) {
+                    sb.append('0')
+                }
+                sb.append(hex)
+            }
+            sb.toString()
+        } catch (e: NoSuchAlgorithmException) {
+            null
+        }
+    }
+
+    /**
+     * Check the device to make sure it has the Google Play Services APK.
+     *
+     *
+     * If it doesn't, display a dialog that allows users to download the APK from the Google Play
+     * Store or enable it in the device's system settings.
+     *
+     * @param context non null activity, used to display update dialog if necessary.
+     * @return true if play services are available, false otherwise
+     */
+    fun checkPlayServices(context: Activity): Boolean {
```

Comment:
Can we make this as extension of `Context` class?

Reply:
updated